[Cire] Step-3 DB를 활용한 저장소 기능 구현,  배포 완료 및 학습계획서 작성

cire0304 opened the pull request:
# 좋았던 점

지루하게 반복되던 코드를, JDBC 템플릿을 사용하여 깔끔하게 정리가 가능하다는게 너무 좋았다.
역시 개발자는 반복되거나, 하기 싫은 일은 하지 않기위해 다양한 방법들을 미리 만들어 두는 부지러한 게으름뱅이라는 것을 느낄 수 있어서 재밌었다.

또한 인생 처음 배포라는 것을 해보았고, 배포에도 쉽고 빠른 방법들이 있고, 그보다 더 빠른 방법들이 있다는 것을 알 수 있는 하루였다.
(로컬에서 build 후, jar 파일만 scp 로 전송하는 방법등...)

나의 첫 서버 : http://15.164.169.12:8080/

# 해야할 것

1. 로그인 기능을 구현하기위한 DB 테이블 수정
2. 로그인 기능 구현하기

# 아쉬웠던 점

기능 구현에 급급하여, 검증 및 예외처리 부분을 신경을 쓰지 못한점이 너무 아쉬웠다.

이번 주 미션은 인증 및 로그인 기능을 구현하는 아주 중요한 미션이므로, 기능 구현에 충실하고,
코드의 질을 높이는 작업은 주말에 하도록 하자. (사실 순서가 반댄가? ㅎㅎ;)

cire0304 committed:
Feat : H2 데이터 베이스를 활용한 저장소 기능 구현

H2 데이터 베이스를 이용한, article 저장소 및 user 저장소 기능 구현

cire0304 committed:
Config : 데이터 베이스 라이브러리 주입 및 기타 설정 변경

cire0304 committed:
Refactor : OCP 원칙 준수하도록 코드 수정

article 저장소 및 user 저장소의 인터페이스를 만들었다.
그 후, 모든 스프링 빈이 구현체가 아닌 인터페이스를 의존하도록 코드를 수정하였다.

honux77 commented:
앞으로 ORM이나 다른 기술을 배워서 클리한 코드를 만들고 싶다.  --> SQL 을 모르면 ORM을 잘 쓸 수 없어요 ㅎㅎ.

honux77 commented on src/main/java/kr/codesqaud/cafe/config/repositoryConfig/ConnectionConfig.java:
```diff
@@ -0,0 +1,7 @@
+    public static final String URL = "jdbc:h2:tcp://localhost/~/test";
+    public static final String USERNAME = "sa";
+    public static final String PASSWORD = "";
+}
```
EOL 자동 설정을 안 했군요.

honux77 replied:
h2에서 JDBC  DataSource 설정은 다른 방법도 가능할 거예요. 다시 확인해 보세요.

honux77 commented on src/main/java/kr/codesqaud/cafe/controller/UserController.java:
```diff
@@ -27,30 +29,30 @@ public String create(@RequestParam String userId,
     public String profile(@PathVariable String userId,
                           Model model) {
-        User user = memoryUserRepository.findUser(userId); // null 포인트 예외 처리 필요
+        User user = userRepository.findUser(userId); // null 포인트 예외 처리 필요
```
//TODO: 
를 활용해 보세요.

honux77 commented on src/main/java/kr/codesqaud/cafe/repository/h2Repository/H2ArticleRepository.java:
```diff
@@ -0,0 +1,162 @@
+public class H2ArticleRepository implements ArticleRepository {
+    private static final Logger logger = LoggerFactory.getLogger("article database");
+
+    @Autowired
+    public H2ArticleRepository()  {}
```
필요한 부분일까요?

honux77 reviewed (changes requested):
JDBC가 아니고 Spring JDBC를 사용해서 구현해 보세요. 
변경 요청 보냈습니다.

cire0304 committed:
fix : 불필요한 요청 제거

불필요한 리다이렉트 응답 제거

cire0304 committed:
DOC : 학습 계획 작성

cire0304 committed:
style : 쓸데없는 주석 제거

cire0304 committed:
Refactor : 저장소 관련 클래스 리펙토링

스프링 에서 제공하는 유틸 라이브러리를 사용하여, 깔끔한 코드로 리펙토링하였다.

cire0304 committed:
HTML : HTML 수정

잘못된 문구 수정

cire0304 committed:
config : 서버용으로 코드 수정

cire0304 committed:
config : apprication properties 변경

cire0304 committed:
fix : 잘못된 뷰 템플릿 경로  변경

로컬 어플리케이션에선 뷰 탬플릿 경로를 잘 잡아 주었지만,
서버 어플리케이션에선 경로를 못잡아서, 뷰 템플릿 경로 변경

cire0304 committed:
Rename : 메소드 이름 변경

cire0304 committed:
refactor : JDBC 탬플릿 사용도록 리펙토링

기존의 순수한 자바코드를 사용하여 DB에 접근하기 때문에 코드가 많이 지져분했다.
JDBC 탬플릿 코드를 사용하여, 코드를 클린하게 만들었다.

cire0304 committed:
Doc : README 수정

cire0304 committed:
Remove : 사용되지 않는 라이브러리 제거

honux77 reviewed (approved):
수고하셨습니다.
